packages/wallet-ui/src/components/ui/molecule/AddressInput/AddressInput.view.tsx

```diff
@@ -22,20 +22,32 @@ import { STARKNET_ADDRESS_LENGTH } from 'utils/constants';
 interface Props extends InputHTMLAttributes<HTMLInputElement> {
   label?: string;
   setIsValidAddress?: Dispatch<SetStateAction<boolean>>;
+  disableValidate?: boolean;
+  validateError?: string;
+  validInput?: boolean;
 }
 
 export const AddressInputView = ({
   disabled,
   onChange,
   label,
   setIsValidAddress,
+  disableValidate,
+  validateError,
+  validInput,
   ...otherProps
 }: Props) => {
   const [focused, setFocused] = useState(false);
   const inputRef = useRef<HTMLInputElement>(null);
   const [error, setError] = useState('');
   const [valid, setValid] = useState(false);
 
+  useEffect(() => {
```

Comment:
nit: 
i think we can simplify a bit, so we can get rid of double validation

as valid is true when
- has no error
- value is not empty

the tradeoff is the icon may show "tick" if the fetching is in process, but if the stark name is not correct, it will turn back to "cross"

```typescript
useEffect(() => {
    if (!disableValidate || !inputRef.current) return;
    setValid(inputRef.current.value !== '' && validateError === '');
    setError(validateError ?? '');
  }, [disableValidate, validateError, inputRef]);

```